ray_write_shapefiles.py

```diff
@@ -0,0 +1,169 @@
+
+    def get_coordinate_system_info(self, file_path: str, file_bytes: bytes):
+        try:
+            # Open the dataset
+            # Create virtual file system file for image to use GDAL's file apis.
+            vfs = gdal_vfs.GDALVirtualFileSystem(
+                file_path, file_bytes)
+            virtual_file_path = vfs.create_virtual_file()
+            print("here is the virtual file path: ", virtual_file_path)
+            dataset = gdal.Open(virtual_file_path)
+
+            # Check if the dataset is valid
+            if dataset is None:
+                print("gdal error: ", gdal.GetLastErrorMsg())
+                raise Exception("Error: Unable to open the dataset")
+
+            # Get the spatial reference
+            spatial_ref = osr.SpatialReference()
+
+            # Try to import the coordinate system from WKT
+            if spatial_ref.ImportFromWkt(dataset.GetProjection()) != gdal.CE_None:
+                raise Exception(
+                    "Error: Unable to import coordinate system from WKT.")
+
+            # Check if the spatial reference is valid
+            if spatial_ref.Validate() != gdal.CE_None:
+                raise Exception("Error: Invalid spatial reference.")
+
+            # Export the spatial reference to WKT
+            dataset = None
+            vfs.close_virtual_file()
+            return spatial_ref.ExportToWkt()
+
+        except Exception as e:
+            print(f"Error when getting the coordinate system info: {e}")
+            dataset = None
+            vfs.close_virtual_file()
+            return None
+
+    def write_prj_file(self, geotiff_path: str, geotiff_bytes: bytes, prj_file_path: str):
+        """
+        Will create the prj file by getting the geo cordinate system from the input tiff file
+
+        Parameters
+        ----------
+        geotiff_path : geotiff_path : Path to the geo tiff used for processing
+        geotiff_bytes: Bytes of the input image
+        prj_file_path : Path to the location to create the prj files
+        """
+        try:
+            # Get the coordinate system information
+            wkt = self.get_coordinate_system_info(geotiff_path, geotiff_bytes)
+
+            print(wkt)
+            if wkt is not None:
+                # Write the WKT to a .prj file
+                if self.config.GCP_FILESYSTEM is not None:
+                    with self.config.GCP_FILESYSTEM.open(prj_file_path, "w") as prj_file:
+                        prj_file.write(wkt)
+                else:
+                    with open(prj_file_path, "w") as prj_file:
+                        prj_file.write(wkt)
+                print(
+                    f"Coordinate system information written to {prj_file_path}")
+            else:
+                print("Failed to get coordinate system information.")
+
+        except Exception as e:
+            print(f"Error when trying to write the prj file: {e}")
+
+    def write_shapefile_helper(self, row: Dict[str, Any], w):
```

Comment:
nit: type hint for "w" would be very helpful!

Reply:
Also, this seems to be a private function of the class. Although you can't enforce private functions, it is common practice to prefix private function names with "__" to indicate to others that they shouldn't be calling this function from outside the class.

Reply:
good call!

Reply:
honestly there are a lot of private functions that should have the __ prefix